src/plugin.js

```diff
@@ -93,15 +93,31 @@ const initPlugin = function(player, options) {
   // creates and tracks a player listener if the player looks alive
   const healthcheck = function(type, fn) {
     let check = function() {
-      // playback isn't expected if the player is paused, shut
-      // down monitoring
-      if (player.paused()) {
-        return resetMonitor();
-      }
-      // playback isn't expected once the video has ended
-      if (player.ended()) {
-        return resetMonitor();
+      if (!player.error()) {
+        // error if using Flash and its API is unavailable
+        let tech = player.$('.vjs-tech');
+
+        if (tech &&
+            tech.type === 'application/x-shockwave-flash' &&
+            !tech.vjs_getProperty) {
+          player.error({
+            code: -2,
+            type: 'PLAYER_ERR_TIMEOUT'
+          });
+          return;
+        }
+
+        // playback isn't expected if the player is paused, shut
+        // down monitoring
```

Comment:
I think this comment is somewhat inaccurate. Technically, you're saying "keep checking, it looks like everything is fine right now".